sklearn/ensemble/forest.py

```diff
@@ -417,6 +418,8 @@ def _set_oob_score(self, X, y):
         self.oob_score_ = oob_score / self.n_outputs_
 
     def _validate_y_class_weight(self, y):
+        assert_non_regression_targets(y)
```

Comment:
`check` functions seem to raise exceptions too, specifically `check_array`. I like `check_non_regression_target` better than `assert_non_regression_target`, personally.